ibis/backends/sql/compilers/base.py

```diff
@@ -671,12 +671,28 @@ def fn(node, __unused__, **kwargs):
         elif isinstance(out, sge.Subquery):
             out = out.this
 
-        # add cte definitions to the select statement
+        merged_ctes = []
         for cte in ctes:
-            alias = sg.to_identifier(aliases[cte], quoted=self.quoted)
-            out = out.with_(
-                alias, as_=results[cte].this, dialect=self.dialect, copy=False
+            this = results[cte]
+            if "alias" in results[cte].args:
```

Comment:
```suggestion
            if "alias" in this.args:
```

since you've already defined `this` on the previous line.

Reply:
Whoops! Fixed!